__tests__/app.test.js

```diff
@@ -5,32 +5,143 @@ const app = require("../app");
+        expect(response.body.api).toEqual(endpointJson);
+      });
+  });
+});
+
+describe("GET /api/articles/:article_id", () => {
+  test("should return a 200 status code", () => {
+    return request(app).get("/api/articles/1").expect(200);
+  });
+
+  test("should return a 404 status code for an article that doesn't exist", () => {
+    const nonExistentArticleId = 76;
+
+    return request(app)
+      .get(`/api/articles/${nonExistentArticleId}`)
+      .expect(404);
+  });
+
+  test("should return the article details for an existing article", () => {
+    const existingArticleId = 6;
+
+    return request(app)
+      .get(`/api/articles/${existingArticleId}`)
+      .expect(200)
+      .then((response) => {
+        expect(Array.isArray(response.articles)).toBe(false);
+        expect(response.body.articles[0]).toHaveProperty("author");
+        expect(response.body.articles[0]).toHaveProperty("title");
+        expect(response.body.articles[0]).toHaveProperty(
+          "article_id",
+          existingArticleId
+        );
+        expect(response.body.articles[0]).toHaveProperty("body");
+        expect(response.body.articles[0]).toHaveProperty("topic");
+        expect(response.body.articles[0]).toHaveProperty("created_at");
+        expect(response.body.articles[0]).toHaveProperty("votes");
+        expect(response.body.articles[0]).toHaveProperty("article_img_url");
       });
   });
 });
-//Checks that it returns an array
+
+test("should return the article details for an existing article", () => {
+  const anotherExistingArticleId = 1;
+
+  return request(app)
+    .get(`/api/articles/${anotherExistingArticleId}`)
+    .expect(200)
+    .then((response) => {
+      expect(Array.isArray(response.articles)).toBe(false);
+    });
+});
+
+describe("GET /api/articles/", () => {
+  test("should return a 200 status code", () => {
+    return request(app).get("/api/articles/").expect(200);
+  });
+
+  test("should return an array", () => {
+    return request(app)
+      .get("/api/articles/")
+      .expect(200)
+      .expect((response) => {
+        expect(Array.isArray(response.body.articles)).toBe(true);
+        expect(response.body.articles.length).toBeGreaterThan(0);
+        response.body.articles.forEach((article) => {
+          expect(typeof article).toBe("object");
+        });
+      });
+  });
+});
+
+describe("GET /api/articles/:article_id/comments", () => {
+  const anotherExistingArticleId = 1;
+
+  test("should return a 200 status code", () => {
+    return request(app)
+      .get(`/api/articles/${anotherExistingArticleId}/comments`)
+      .expect(200);
+  });
+
+  test("should return the article comment details for an existing article", () => {
+    return request(app)
+      .get(`/api/articles/${anotherExistingArticleId}/comments`)
+      .expect(200)
+      .then((response) => {
+        console.log(response.body.comments);
+        expect(Array.isArray(response.body.comments)).toBe(true);
```

Comment:
more detail needed here... what do the comments for this article look like? how many should there be?